SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h

```diff
@@ -149,11 +149,17 @@ class SPATIALGDK_API USpatialNetDriver : public UIpNetDriver
 
 	void DelayedSendDeleteEntityRequest(Worker_EntityId EntityId, float Delay);
 
+	bool IsEntityListening(Worker_EntityId EntityId) const;
+	void RegisterListeningEntity(Worker_EntityId EntityId);
+	void UnregisterListeningEntity(Worker_EntityId EntityId);
+
 private:
 	TUniquePtr<FSpatialOutputDevice> SpatialOutputDevice;
 
 	TMap<Worker_EntityId_Key, USpatialActorChannel*> EntityToActorChannel;
 
+	TSet<int64> ListeningEntities;
```

Comment:
use `TSet<Worker_EntityId>`

Reply:
It fails to compile on Linux, since `Worker_EntityId` is different from types supported by `GetTypeHash` function. Does it worth adding `GetTypeHash(Worker_EntityId)` to UE4 to be able to use `TSet<Worker_EntityId>`?

Reply:
Ah that actually makes some sense. I forgot unreal has it's own odd hashing, I'm fairly sure std::hash would correctly pick it up :(.
Yes I think we should add unreal hash functions for SpatialOS types in that case.

Reply:
In this case it would have to be added to global scope which I'm not thrilled about but I don't think there is a way around that. Except providing a hash function inside the class owning the set maybe. But it seems better to just make hashing on improbable types work.

Reply:
If someone wants to argue this is a temporary fix for single worker and we won't even be keeping that set around forever then fine. Although I suspect that in general we will want to use these types as keys.

Reply:
We've a typedef for this - Worker_EntityId_Key

Reply:
@m-samiec That's great! Will use this one.

Reply:
Wait that seems weird. Why not just define a hash function?

Reply:
Changed it to `Worker_EntityId_Key` for now. However, we can discuss it further.